analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -25,10 +25,31 @@ public sealed class ObjectsShouldNotBeDisposedMoreThanOnce : ObjectsShouldNotBeD
     public static readonly DiagnosticDescriptor S3966 = DescriptorFactory.Create(DiagnosticId, MessageFormat);
     protected override DiagnosticDescriptor Rule => S3966;
 
-    public override bool ShouldExecute() => true;
+    public override bool ShouldExecute()
+    {
+        var walker = new Walker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
 
     protected override bool IsDispose(IMethodSymbol method) =>
         method.IsIDisposableDispose()
         || method.IsIAsyncDisposableDisposeAsync()
         || method.ExplicitInterfaceImplementations.Any(x => x.IsIDisposableDispose() || x.IsIAsyncDisposableDisposeAsync());
+
+    private sealed class Walker : SafeVisualBasicSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!Result)
+            {
+                base.Visit(node);
+            }
+        }
+
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node) =>
```

Comment:
I'm afraid that this will need to be also on member access, because you can have `d.Dispose` without parenthesis, and that's not an invocation

Reply:
@pavel-mikula-sonarsource But[ in the rule we inspect only invocations](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ObjectsShouldNotBeDisposedMoreThanOnceBase.cs#LL35C25-L35C25).

In case I should add a UT to document this case and open an issue to support that - for now, it does not raise if the invocation happens through member access only.

Reply:
Add test case to cover ShouldExecute and you'll see it works.

What syntactically looks like member access, to make dev life easy, is represented as invocation in MSIL and also CFG and operations. So it will produce InvocationOperation and it's fine.

When I say that VB is dev-first language, it does not apply to analyzer developers :D

Reply:
The override still needs to change from invocation to member access